File: internal/models/providers.go
package models

var ProviderModels = map[string][]string{
	"openai": {
		"gpt-4o", "gpt-4o-mini", "gpt-4-turbo", "gpt-4", "gpt-3.5-turbo",
		"o1", "o1-mini", "o1-preview", "o3-mini",
	},
	"anthropic": {
		"claude-3-5-sonnet-20241022", "claude-3-5-haiku-20241022",
		"claude-3-opus-20240229", "claude-3-sonnet-20240229", "claude-3-haiku-20240307",
	},
	"deepseek": {
		"deepseek-chat", "deepseek-reasoner", "deepseek-coder",
	},
	"qwen": {
		"qwen-max", "qwen-max-longcontext", "qwen-plus", "qwen-turbo",
		"qwen-long", "qwen-vl-max", "qwen-vl-plus",
	},
	"zhipu": {
		"glm-4-plus", "glm-4-0520", "glm-4", "glm-4-air", "glm-4-airx",
		"glm-4-flash", "glm-4v", "glm-4v-plus",
	},
	"moonshot":   {"moonshot-v1-8k", "moonshot-v1-32k", "moonshot-v1-128k"},
	"minimax":    {"abab6.5s-chat", "abab6.5g-chat", "abab6.5-chat", "abab5.5-chat"},
	"baichuan":   {"Baichuan4", "Baichuan3-Turbo", "Baichuan3-Turbo-128k"},
	"volcengine": {"doubao-pro-4k", "doubao-pro-32k", "doubao-pro-128k"},
	"ernie":      {"ernie-4.0-8k", "ernie-3.5-8k", "ernie-speed-8k"},
	"yi":         {"yi-lightning", "yi-large", "yi-medium", "yi-spark"},
	"google":     {"gemini-1.5-pro", "gemini-1.5-flash", "gemini-pro"},
	"mistral":    {"mistral-large-latest", "mistral-medium", "codestral-latest"},
	"hunyuan":    {"hunyuan-lite", "hunyuan-standard", "hunyuan-pro"},
	"spark":      {"spark-v3.5", "spark-v3.0"},
}

var ProviderBaseURLs = map[string]string{
	"openai":     "https://api.openai.com/v1",
	"anthropic":  "https://api.anthropic.com/v1",
	"deepseek":   "https://api.deepseek.com",
	"qwen":       "https://dashscope.aliyuncs.com/compatible-mode/v1",
	"zhipu":      "https://open.bigmodel.cn/api/paas/v4",
	"moonshot":   "https://api.moonshot.cn/v1",
	"minimax":    "https://api.minimax.chat/v1",
	"baichuan":   "https://api.baichuan-ai.com/v1",
	"volcengine": "https://ark.cn-beijing.volces.com/api/v3",
	"ernie":      "https://aip.baidubce.com/rpc/2.0/ai_custom/v1",
	"yi":         "https://api.lingyiwanwu.com/v1",
	"google":     "https://generativelanguage.googleapis.com/v1beta",
	"mistral":    "https://api.mistral.ai/v1",
	"hunyuan":    "https://hunyuan.tencentcloudapi.com",
	"spark":      "https://spark-api-open.xf-yun.com/v1",
}

func GetProviderModels(provider string) []string {
	return ProviderModels[provider]
}

func GetProviderBaseURL(provider string) string {
	return ProviderBaseURLs[provider]
}

func IsProviderSupported(provider string) bool {
	_, ok := ProviderModels[provider]
	return ok
}

func GetAllProviders() []string {
	providers := make([]string, 0, len(ProviderModels))
	for p := range ProviderModels {
		providers = append(providers, p)
	}
	return providers
}

type AccountRecord struct {
	ID           string `json:"id"`
	Provider     string `json:"provider"`
	APIKey       string `json:"api_key"`
	Priority     int    `json:"priority"`
	Enabled      bool   `json:"enabled"`
	QuotaLimit   int64  `json:"quota_limit"`
	QuotaUsed    int64  `json:"quota_used"`
	QuotaResetAt string `json:"quota_reset_at,omitempty"`
	CreatedAt    string `json:"created_at"`
	UpdatedAt    string `json:"updated_at"`
}

type ModelScoreRecord struct {
	Model        string `json:"model"`
	Provider     string `json:"provider"`
	QualityScore int    `json:"quality_score"`
	SpeedScore   int    `json:"speed_score"`
	CostScore    int    `json:"cost_score"`
	Enabled      bool   `json:"enabled"`
	IsCustom     bool   `json:"is_custom"`
	CreatedAt    string `json:"created_at"`
	UpdatedAt    string `json:"updated_at"`
}

type RouterConfigRecord struct {
	DefaultStrategy string `json:"default_strategy"`
	DefaultModel    string `json:"default_model"`
	UseAutoMode     bool   `json:"use_auto_mode"`
}

type APIKeyRecord struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Key         string `json:"key"`
	Permissions string `json:"permissions"`
	Enabled     bool   `json:"enabled"`
	LastUsedAt  string `json:"last_used_at,omitempty"`
	CreatedAt   string `json:"created_at"`
	ExpiresAt   string `json:"expires_at,omitempty"`
}

type UserRecord struct {
	Username     string `json:"username"`
	PasswordHash string `json:"password_hash"`
	Role         string `json:"role"`
	Email        string `json:"email,omitempty"`
	CreatedAt    string `json:"created_at"`
	UpdatedAt    string `json:"updated_at"`
}

type FeedbackRecord struct {
	ID         string `json:"id"`
	RequestID  string `json:"request_id"`
	Model      string `json:"model"`
	Provider   string `json:"provider"`
	TaskType   string `json:"task_type"`
	Rating     int    `json:"rating"`
	Comment    string `json:"comment"`
	LatencyMs  int64  `json:"latency_ms"`
	TokensUsed int64  `json:"tokens_used"`
	CacheHit   bool   `json:"cache_hit"`
	CreatedAt  string `json:"created_at"`
}

File: web/src/constants/pages/providers.ts
export const PROVIDERS_AVAILABLE_MODELS = [
  'gpt-4o', 'gpt-4o-mini', 'gpt-4-turbo', 'gpt-4', 'gpt-3.5-turbo', 'o1', 'o1-mini', 'o1-preview',
  'claude-3-5-sonnet-20241022', 'claude-3-opus-20240229', 'claude-3-sonnet-20240229', 'claude-3-haiku-20240307',
  'gemini-2.0-flash-exp', 'gemini-1.5-pro', 'gemini-1.5-flash', 'gemini-pro',
  'gpt-4o', 'gpt-4', 'gpt-35-turbo',
  'doubao-pro-256k', 'doubao-pro-128k', 'doubao-pro-32k', 'doubao-lite-128k', 'doubao-lite-32k',
  'qwen-max', 'qwen-max-longcontext', 'qwen-plus', 'qwen-turbo', 'qwen-long',
  'ernie-4.0-8k', 'ernie-4.0', 'ernie-3.5-8k', 'ernie-3.5', 'ernie-speed-8k', 'ernie-speed',
  'glm-4-plus', 'glm-4-0520', 'glm-4-air', 'glm-4-airx', 'glm-4-long', 'glm-4-flash',
  'hunyuan-lite', 'hunyuan-standard', 'hunyuan-pro', 'hunyuan-turbo',
  'moonshot-v1-8k', 'moonshot-v1-32k', 'moonshot-v1-128k',
  'abab6.5-chat', 'abab6.5s-chat', 'abab5.5-chat', 'abab5.5s-chat',
  'Baichuan4', 'Baichuan3-Turbo', 'Baichuan3-Turbo-128k', 'Baichuan2-Turbo',
  'spark-v3.5', 'spark-v3.0', 'spark-v2.0', 'spark-v1.5',
  'pangu-natural-language-10b', 'pangu-nlg-2b',
  'nova-ptc-xl-v1', 'nova-ptc-large-v1',
  '360gpt2-pro', '360gpt-turbo',
  'deepseek-chat', 'deepseek-reasoner'
] as const

export const PROVIDERS_COLOR_MAP: Record<string, string> = {
  openai: '#10A37F',
  azure: '#0078D4',
  anthropic: '#CC785C',
  google: '#4285F4',
  volcengine: '#FF4D4F',
  qwen: '#FF6A00',
  ernie: '#2932E1',
  zhipu: '#3657ED',
  hunyuan: '#00A3FF',
  moonshot: '#1A1A1A',
  minimax: '#615CED',
  baichuan: '#0066FF',
  spark: '#E60012',
  deepseek: '#4D6BFE',
  custom: '#8B5CF6'
}

export const PROVIDERS_ICON_MAP: Record<string, string> = {
  openai: 'ChatDotRound',
  azure: 'Platform',
  anthropic: 'ChatLineRound',
  google: 'Star',
  volcengine: 'Lightning',
  qwen: 'Sunny',
  ernie: 'Reading',
  zhipu: 'MagicStick',
  hunyuan: 'Connection',
  moonshot: 'Moon',
  minimax: 'Cpu',
  baichuan: 'TrendCharts',
  spark: 'Promotion',
  deepseek: 'Search',
  custom: 'Setting'
}

export const PROVIDERS_ENDPOINT_MAP: Record<string, string> = {
  openai: 'https://api.openai.com/v1',
  azure: 'https://your-resource.openai.azure.com',
  anthropic: 'https://api.anthropic.com/v1',
  google: 'https://generativelanguage.googleapis.com/v1beta',
  volcengine: 'https://ark.cn-beijing.volces.com/api/v3',
  qwen: 'https://dashscope.aliyuncs.com/api/v1',
  ernie: 'https://aip.baidubce.com/rpc/2.0/ai_custom/v1',
  zhipu: 'https://open.bigmodel.cn/api/paas/v4',
  hunyuan: 'https://hunyuan.tencentcloudapi.com',
  moonshot: 'https://api.moonshot.cn/v1',
  minimax: 'https://api.minimax.chat/v1',
  baichuan: 'https://api.baichuan-ai.com/v1',
  spark: 'https://spark-api-open.xf-yun.com/v1',
  deepseek: 'https://api.deepseek.com/v1'
}
